core/src/main/java/org/apache/accumulo/core/client/admin/TableOperations.java

```diff
@@ -630,28 +631,41 @@ void load()
   /**
    * @since 2.0.0
    */
-  interface ImportExecutorOptions extends ImportSourceOptions {
+  interface ImportDestinationOptions extends ImportFinalOptions {
+
+    /**
+     * Load files in the directory to the row ranges specified in the plan. The plan should contain
+     * at least one entry for every file in the directory.
+     */
+    ImportFinalOptions usingPlan(LoadPlan service);
+
+    // The javadoc below intentionally use a fully qualified class name in the value tag, otherwise
+    // it would not render properly.
     /**
      * Files are examined to determine where to load them. This examination is done in the current
-     * process using multiple threads. If this property is not set, then the client property
-     * {@code bulk.threads} is used to create a thread pool.
+     * process using multiple threads. If this method is not called, then the client property
+     * {@code bulk.threads} is used to create a thread pool. This property defaults to
+     * {@value org.apache.accumulo.core.conf.ClientDefaults#BULK_LOAD_THREADS_DEFAULT}.
      *
      * @param service
      *          Use this executor to run file examination task
      * @return ImportSourceOptions
      */
-    ImportSourceOptions usingExecutor(Executor service);
+    ImportFinalOptions examiningWith(Executor service);
```

Comment:
The reason I used examining was because I thought it read nicely in a fluent chain.

```
conn.tableOperations().addFilesTo(tableName).from(dir).examiningWith(service).load();
```

Reply:
If variations of "examine" don't work, we could try variations of "inspect" or some other synonym.

Reply:
I changed this back to `usingExecutor` because its more consistent with the other method prefixes in the type.  However I may change it back to some variation of examine or inspect.